ckanext/scheming/plugins.py

```diff
@@ -69,65 +51,68 @@ class _SchemingMixin(object):
-            'scheming_get_preset': helpers.scheming_get_preset,
-            'scheming_get_schema': helpers.scheming_get_schema,
-            'scheming_get_timezones': helpers.scheming_get_timezones,
-            'scheming_datetime_to_tz': helpers.scheming_datetime_to_tz,
-            'scheming_datastore_choices': helpers.scheming_datastore_choices,
-            'scheming_display_json_value': helpers.scheming_display_json_value,
-            }
+
+        return dict(
+            inspect.getmembers(
+                helpers,
+                lambda o: inspect.isfunction(o) and o.__name__.startswith(
+                    'scheming_')
+            )
+        )
 
     def get_validators(self):
         if _SchemingMixin._validators_loaded:
             return {}
         _SchemingMixin._validators_loaded = True
-        return {
-            'scheming_choices': scheming_choices,
-            'scheming_required': scheming_required,
-            'scheming_multiple_choice': scheming_multiple_choice,
-            'scheming_multiple_choice_output': scheming_multiple_choice_output,
+
+        validators = dict(
+            inspect.getmembers(
+                validation,
+                lambda o: inspect.isfunction(o) and o.__name__.startswith(
+                    'scheming_')
+            )
+        )
+
+        validators.update({
             'convert_to_json_if_date': convert_to_json_if_date,
-            'convert_to_json_if_datetime': convert_to_json_if_datetime,
-            'scheming_isodatetime': scheming_isodatetime,
-            'scheming_isodatetime_tz': scheming_isodatetime_tz,
-            'scheming_valid_json_object': scheming_valid_json_object,
-            'scheming_load_json': scheming_load_json,
-            }
+            'convert_to_json_if_datetime': convert_to_json_if_datetime
+        })
+
+        return validators
 
     def _add_template_directory(self, config):
         if _SchemingMixin._template_dir_added:
             return
         _SchemingMixin._template_dir_added = True
         add_template_directory(config, 'templates')
+        add_resource('fanstatic', 'scheming')
 
-    def _load_presets(self, config):
+    @staticmethod
+    def _load_presets(config):
         if _SchemingMixin._presets is not None:
             return
-        presets = config.get('scheming.presets', DEFAULT_PRESETS).split()
-        _SchemingMixin._presets = {}
-        for f in reversed(presets):
-            for pp in _load_schema(f)['presets']:
-                _SchemingMixin._presets[pp['preset_name']] = pp['values']
+
+        presets = reversed(
+            config.get(
+                'scheming.presets',
+                DEFAULT_PRESETS
+            ).split()
+        )
+
+        _SchemingMixin._presets = {
+            field['preset_name']: field['values']
+            for preset_path in presets
+            for field in _load_schema(preset_path)['presets']
+        }
```

Comment:
I like renaming `f` to `preset_path` but otherwise I find the original code easier to understand

Reply:
There was a reason for this, to make the assignment a single atomic operation...can't remember why tho, I think something I was experimenting with in load_schema would call something else that would in turn look at the presets again...

Reply:
ok, so we could build this as a new dict then overwrite the shared one to make it atomic.

The way these values are cached in class variables probably isn't safe even with the atomic update for example with tests reloading configurations. Need a better way to store and update these.